### PR TITLE
Update to the new Search Console API version

### PR DESCRIPTION
**Pull request by benbowler**
## Summary

Addresses issue #2522

## Relevant technical choices

As mentioned in [a comment on the original issue](https://github.com/google/site-kit-wp/issues/2522#issuecomment-752991239),  the `dataState` property is [only available on `master` branch](https://github.com/googleapis/google-api-php-client-services/blob/master/src/Google/Service/SearchConsole/SearchAnalyticsQueryRequest.php#L22) not [on the v0.156 release](https://github.com/googleapis/google-api-php-client-services/blob/v0.156/src/Google/Service/SearchConsole/SearchAnalyticsQueryRequest.php) as of yet so I've implemented this with the `dev-master` branch of the repository in `composer.json`.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: composer.json
Comment (aaemnnosttv) [inline composer.json:21]: This is probably okay to change since this `composer.json` is more of a dev-dependency (you can't install a fully-working version of the plugin by requiring `google/google-site-kit`).

It's a bit odd that the new version hasn't been tagged yet but 🤷‍♂️ 

I'm okay with this change if @felixarntz is (we control when its updated anyways) but I would prefer we used a version constraint instead; I understand the latest tag doesn't include `dataState` yet.

**Review by aaemnnosttv — APPROVED** [review]
LGTM but asking @felixarntz to review the version constraint change to `google/apiclient-services`

File: composer.json
Comment (felixarntz) [reply composer.json:21]: Good catch - I hadn't noticed that the change hasn't been released yet. Let's roll this change here back and wait for the next release instead. I've just emailed the maintainer of the PHP library and will keep an eye out for the update. We can have this issue wait in MR until then.

File: composer.lock
Comment (felixarntz) [inline composer.lock:75]: See above, let's revert the `composer.lock` changes.

File: includes/Modules/Search_Console.php
Comment (felixarntz) [inline includes/Modules/Search_Console.php:543]: While we're at it, let's add a `since` annotation here, which was previously missing. Since it's a new method though, we can use the `n.e.x.t` tag.
```suggestion
	 * Get the configured SearchConsole service instance.
	 *
	 * @since n.e.x.t
```

**Review by felixarntz — CHANGES_REQUESTED** [review]
Awesome work! LGTM, just some minor things below, TLDR we should wait for the library update.

File: composer.lock
Comment (benbowler) [reply composer.lock]: Reverted.

File: composer.json
Comment (felixarntz) [inline composer.json:21]: @benbowler @aaemnnosttv Shouldn't this be `>0.156` instead? Not a big deal for now though - alternatively we can wait for the new version to have the actual version number and then use `>=` with that one instead.

**Review by felixarntz — CHANGES_REQUESTED** [review]
@benbowler I've updated the `composer.json` here to use the new 0.157 or higher, which was released just 2 days ago and includes the necessary updates.

The production code here is good to go, however this PR breaks the Storybook stories for all Search Console widgets (they display the zero data state instead of their regular loaded state) because the fixtures / reference date are still relying on the 2 day offset. Can you update these accordingly? Simplest would probably be to move the reference dates by one day.

**Comment by benbowler** [comment]
The storybooks are now fixed and this is ready for CR and MR.

**Review by felixarntz — APPROVED** [review]
Awesome!

**Pull request merged**
